### PR TITLE
add check to not clone from template if custom cluster bootstrap

### PR DESCRIPTION
**Pull request by maralavi**
### What this PR does / why we need it
Add check to not clone from template if custom cluster bootstrap annotation is present

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
Add check to not clone from template if custom cluster bootstrap annotation is present
```

### CODE REVIEW

**Review by vijaykatam — COMMENTED** [review]
Minor comments.

File: addons/controllers/clusterbootstrap_controller.go [hunk 1]
```diff
@@ -393,6 +393,9 @@ func (r *ClusterBootstrapReconciler) createOrPatchClusterBootstrapFromTemplate(c
 	clusterBootstrapHelper := clusterbootstrapclone.NewHelper(
 		r.context, r.Client, r.aggregatedAPIResourcesClient, r.dynamicClient, r.gvrHelper, r.Log)
 	if clusterBootstrap.UID == "" {
+		if _, ok := cluster.Annotations[constants.CustomClusterBootstrap]; ok {
+			return nil, nil
```
[the remark this answers is not in the document]
Comment (maralavi) [reply addons/controllers/clusterbootstrap_controller.go:398]: Thanks for this, added.

File: addons/README.md [hunk 2]
```diff
@@ -221,8 +221,9 @@ When the config resource is pre-created, addons-controller will use it in lieu o
 #### Pre-create ClusterBootstrap resource
 
 If the use case is to swap out a specific package, for example CNI with your own package then create the ClusterBootstrap object in its entirety
-with the same name as the Cluster resource. Creating this resouce in its entirety can put additional burden on clients so a defaulting webhook is
-provided.
+with the same name as the Cluster resource. In case where a custom ClusterBootstrap object was provided, the ClusterBootstrap object will not be
+cloned from the ClusterBootstrapTemplate anymore. Creating the ClusterBootstrap resource in its entirety can put additional burden on clients so a
+defaulting webhook is provided.
```
Comment (vijaykatam) [inline addons/README.md:227]: I think we also should call out the annotation on the cluster object as a separate mechanism

Comment (maralavi) [reply addons/README.md]: Sure, made this change.

**Review by vijaykatam — APPROVED** [review]
LGTM. Thanks for readme update.

**Pull request merged**